Topic1/key6/testClient.java

```diff
@@ -0,0 +1,19 @@
+package Topic1.key6;
+
+import javax.security.auth.Subject;
+
+
+
+public class testClient {
+
+	public static void main(String[] args) {
+		MailingList ml=new MailingList();
+		IObserver ob=new Manager();
+		
+		ml.attach(ob);
+		ml.setState(new Notification(Notification.PRICE_CHANGED));
+		ml.setState(new Notification(Notification.TRANSACTION_DONE));
```

Comment:
This requeriment is tricky and integration with Cart is missing. 
Since notification to the market manager is through a mailing list, there is no need to model the market manager object at all. We can keep the Manager email in MailingList and notify MailingList about changes in Cart.
Looking at the Observer pattern MailingList/MailingListService will be the Observer and Cart or EcommerceService the subject. Please let me know any comments or questions you might have on skype.